decidim-assemblies/app/controllers/decidim/assemblies/assemblies_controller.rb

```diff
@@ -17,9 +17,29 @@ class AssembliesController < Decidim::Assemblies::ApplicationController
       helper_method :collection, :promoted_assemblies, :assemblies, :stats, :assembly_participatory_processes
 
       def index
-        redirect_to "/404" if published_assemblies.none?
-
         enforce_permission_to :list, :assembly
+
+        respond_to do |format|
+          format.html do
+            redirect_to "/404" if published_assemblies.none?
```

Comment:
Could you improve this and change it to `raise ActionController::RoutingError, "Not Found"` ?

Reply:
Why do we use `public_assemblies` here but `collection` with the JSON response?